money-core/src/main/scala/com/comcast/money/core/Formatters.scala

```diff
@@ -35,4 +35,12 @@ object Formatters {
 
   def toHttpHeader(spanId: SpanId): String =
     HttpHeaderFormat.format(spanId.traceId, spanId.parentId, spanId.selfId)
+
+  def fromB3HttpHeaders(traceId: String, maybeParentSpanId: Option[String], maybeSpanId: Option[String]) = Try {
+    (maybeParentSpanId, maybeSpanId) match {
+      case (Some(ps), Some(s)) => new SpanId(traceId, ps.toLong, s.toLong)
+      case (Some(ps), _) => new SpanId(traceId, ps.toLong)
+      case _ => new SpanId(traceId)
+    }
```

Comment:
toLong can throw a NumberFormatException

Reply:
Eww, good catch.

A little complicated the situation:

1. You may have the TraceId header, but not the parent id or the span id
2. You may have the TraceId header, the parent id header, but not the span id
3. You may have all 3
4. You may have none
5. The parent id header could be invalid
6. The span id header could be invalid

I do think Try is appropriate (we have not upgraded to Either yet).  We do need to encode all those rules though.

The signature should take 3 options (traceId: Option[String], parentId: Option[String], spanId: Option[String]) and return a Try[SpanId].

We should fail if any of the `toLong` fails.  Otherwise, we should always return a Success with a SpanId in it.

I tried to play with an implementation but could not get there today.  It is a little tough to deal with, and perhaps just wrapping the whole thing in a `Try` is good enough.

Reply:
I do not have a much better solution, I think this is good, changing my comment